charley_tran/week_01/arrays.html

```diff
@@ -0,0 +1,46 @@
+<html>
+<script>
+// Homework: The Word Guesser
+// You'll create a simple word guessing game where the user gets infinite tries to guess the word (like Hangman without the hangman, or like Wheel of Fortune without the wheel and fortune).
+
+// Create two global arrays: one to hold the letters of the word (e.g. 'F', 'O', 'X'), and one to hold the current guessed letters (e.g. it would start with '', '', '_' and end with 'F', 'O', 'X').
+// Write a function called guessLetter that will:
+// Take one argument, the guessed letter.
+// Iterate through the word letters and see if the guessed letter is in there.
+// If the guessed letter matches a word letter, changed the guessed letters array to reflect that.
+// When it's done iterating, it should log the current guessed letters ('F__') and congratulate the user if they found a new letter.
+// It should also figure out if there are any more letters that need to be guessed, and if not, it should congratulate the user for winning the game.
+// Pretend you don't know the word, and call guessLetter multiple times with various letters to check that your program works.
+// guessing game
+var wordToGuess = ['t','e','s','t']; //word to guess
+var currentGuessedLetters = ['_','_','_','_']; //guessing panel
+
+
+var checkComplete = function() { // this function exists as if(wordToGuess === currentGuessedLetters) does not define for some reason; reduce the clutter a bit.
+	if (currentGuessedLetters.indexOf('_') == -1) {
+		wordToGuess=wordToGuess.toString();
+		currentGuessedLetters=currentGuessedLetters.toString();
+	}
+}
+
+
+var checkWinner = function() {
+	if (currentGuessedLetters===wordToGuess) {
+		return console.log("Congratulations we have a winner!");
+	}
+}
+var guessLetter	= function(guess) {
+	for (var x = 0; x < wordToGuess.length; x ++) {
+	if (wordToGuess[x] === guess) {
+		console.log("Congratulations you found letter " + guess);
+		currentGuessedLetters[x] = guess;
+		console.log(currentGuessedLetters);
+	}   
+	}		checkComplete();
+		checkWinner();
+}
+
+
```

Comment:
I really like your use of functions and your choice of variable names. It made it easier to understand this solution than some of the others.